### PR TITLE
Implement #17037: Replacement of 'b' and '#' with '♭' and '♯' for staff names imported from mxml

### PR DESCRIPTION
**Pull request by asattely**
Resolves: https://github.com/musescore/MuseScore/issues/17037

This replaces b and # with their unicode equivalents during musicxml import iff:

- It is a capital or lowercase letter A-G, OR a valid note name in Italian or French followed by one of those two characters
- That note name is its own token (i.e. surrounded by whitespace or at the end or beginning of the name)

As is pointed out in a comment on the original issue, there is an import option available to replace (or not replace) fonts. This PR does not override that setting, and doesn't touch font stuff at all.

### CODE REVIEW

**Comment by asattely** [comment]
Updated to deal with transpositions in non-english languages, such as Si♭

**Review by oktophonie — APPROVED** [review]

File: src/engraving/libmscore/instrument.cpp [hunk 1]
```diff
@@ -150,6 +150,18 @@ StaffName::StaffName(const String& xmlText, int pos)
     TextBase::validateText(_name); // enforce HTML encoding
 }
 
+void StaffName::xmlReplaceAccidentals()
```
Comment (RomanPudashkin) [inline src/engraving/libmscore/instrument.cpp:153]: This code should be inside importmxml.cpp (since it's a musicxml-related code)

Comment (asattely) [reply src/engraving/libmscore/instrument.cpp:153]: So, it should still be part of `StaffName` but it should be defined in `importmxml.cpp`? Should I move the `Instrument` method below it as well?

Comment (RomanPudashkin) [reply src/engraving/libmscore/instrument.cpp:153]: No, it should not be part of StaffName. You can declare it directly in importmxml.cpp as a static function and pass StaffName as an argument

File: src/engraving/libmscore/instrument.h [hunk 2]
```diff
@@ -425,6 +425,7 @@ class Instrument
     void setIsPrimary(bool isPrimary);
 
     void updateInstrumentId();
+    void updateNamesForAccidentals();
```
Comment (RomanPudashkin) [inline src/engraving/libmscore/instrument.h:428]: Please remove this line (it is not needed)

File: src/importexport/musicxml/internal/musicxml/importmxml.cpp [hunk 3]
```diff
@@ -53,6 +53,34 @@ static int musicXMLImportErrorDialog(QString text, QString detailedText)
     return errorDialog.exec();
 }
 
+void updateNamesForAccidentals(Instrument* inst)
```
Comment (RomanPudashkin) [inline src/importexport/musicxml/internal/musicxml/importmxml.cpp:56]: **static** void updateNamesForAccidentals(Instrument* inst)

**Pull request closed without merging**